src/patient-registration/input/custom-input/relationship-input.component .tsx

```diff
@@ -0,0 +1,135 @@
+//import styles from './autosuggest.css';
+
+interface RelationshipInputProps {
+  setFieldValue(field: string, value: any, shouldValidate?: boolean): void;
+}
+
+export const RelationshipInput: React.FC<RelationshipInputProps> = ({ setFieldValue }) => {
+  const [relationshipTypesOptions, setRelationshipTypesOptions] = useState([]);
+
+  useEffect(() => {
+    const abortController = new AbortController();
+    getAllRelationshipTypes(abortController).then(({ data }) => {
+      data.results.map(option => {
+        setRelationshipTypesOptions(relationshipTypesOptions => [
+          ...relationshipTypesOptions,
+          { uuid: option.uuid + '-A', display: option.aIsToB },
+        ]);
+        setRelationshipTypesOptions(relationshipTypesOptions => [
+          ...relationshipTypesOptions,
+          { uuid: option.uuid + '-B', display: option.bIsToA },
+        ]);
+      });
+    }, createErrorHandler());
+    return () => abortController.abort();
+  }, []);
+
+  const [suggestions, setSuggestions] = useState([]);
+  const [personName, setPersonName] = useState([]);
+
+  const handleSearch = query => {
+    const abortController = new AbortController();
+    getPerson(abortController, query.value).then(({ data }) => {
+      const suggestions = data.results.map(i => ({
+        id: i.uuid,
+        display: i.display,
+      }));
+      const truncatedSuggestions = suggestions.slice(0, 5);
+
+      setSuggestions(truncatedSuggestions);
+    }, createErrorHandler());
+  };
+
+  return (
+    <div>
+      <FieldArray name="relationships">
+        {fieldArrayProps => {
+          const { push, remove, form } = fieldArrayProps;
+          const { values } = form;
+          const { relationships } = values;
+          return (
+            <main className={styles.field}>
+              {relationships.map((relationship, index) => (
+                <section key={index} className={styles.fieldRow}>
+                  <SelectInput name={`relationships[${index}].type`} options={relationshipTypesOptions} />
+                  <Autosuggest
+                    suggestions={suggestions}
+                    onSuggestionsClearRequested={() => {
+                      setSuggestions([]);
+                    }}
+                    getSuggestionValue={suggestion => {
+                      return suggestion.id;
+                    }}
+                    renderSuggestion={suggestion => {
+                      return <span>{suggestion.display}</span>;
+                    }}
+                    inputProps={{
+                      placeholder: 'Person name',
+                      name: `relationships[${index}].name`,
+                      value: personName[index] ? personName[index] : '',
+                      onChange: (_event, { newValue }) => {
+                        if (index < personName.length) {
+                          const personNameList = personName.splice(index, 1, newValue);
+                          setPersonName(personName);
```

Comment:
Something's wrong here. You're not doing anything with `personNameList`. If it's not too tricky to cover this onChange logic with a test case, I think it would be valuable to do so.

Reply:
See my comment above w/r/t having an array named `personName`. Fixing the naming will probably help clarify the logic.